### PR TITLE
feat(data-exploration): "saved" queries

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

It's hard to understand that by adding a column like `count()`, you can convert the "Event Explorer" scene into a powerful data table.

## Changes

Adds a few saved queries to get people exploring: (gif below)

~Also rearranges the filters above the table to better suit this button.~

Everything's still flagged 😅 

## Out of scope

I'd love for there to be a way to save your own queries and also set a new default for the team... but... let's discuss this. We're awkwardly close to recreating saved insights 🤔 

## How did you test this code?

Clicked around in the table.

### CODE REVIEW

**Comment by Twixes** [comment]
This is increasingly becoming an insight… I also have a suspicion that the "hard to understand that by adding a column you can convert this into a powerful data table" problem would be kind of solved by separating the "Live Events" and "Data Table" use cases.

**Comment by mariusandra** [comment]
> I also have a suspicion that the "hard to understand that by adding a column you can convert this into a powerful data table" problem would be kind of solved by separating the "Live Events" and "Data Table" use cases.

In a "Data Table" usecase you'd still have the same problem: how to go from a list view to an aggregate view. Sometimes you want to see every event, sometimes you want to see an aggregation. Said differently, assuming no 

> This is increasingly becoming an insight… 

Yeah, agreed. 

However, just slapping this entire table under "insights" right now seems premature. We can always move this around in a few PRs, but database changes are hard. So let's think it through. I feel there are some missing pieces, and work soon enough will wrap all of them up:

- There's an issue about [renaming "Insights" into "Product Analytics"](https://github.com/PostHog/posthog/issues/13742)
- [PostHog 3000](https://github.com/orgs/PostHog/projects/78/views/1) has a "notebook mode" design idea. When it'll be built is unknowon.
- We'd like data exploration nodes on dashboards. I wouldn't want to now add a new insight type, but this should be just a different node in a unified system.
- It might be better (or at least interesting) to start an exploration from the "data table" view, and convert to a visualisation automatically when you have any aggregations. Plus add a horizontal time axis if you include the timestamp.

**Comment by mariusandra** [comment]
I fixed a test bug, and moved the table header back to have the smallest possible change in this PR:

File: frontend/src/queries/nodes/DataNode/dataNodeLogic.ts [hunk 1]
```diff
@@ -203,6 +203,8 @@ export const dataNodeLogic = kea<dataNodeLogicType>([
                             if (firstTimestamp) {
                                 const nextQuery: EventsQuery = { ...query, after: firstTimestamp }
                                 return nextQuery
+                            } else {
+                                return query
```
Comment (mariusandra) [inline frontend/src/queries/nodes/DataNode/dataNodeLogic.ts:207]: This fixes a bug: before, if there were no entries in the initial response, the "automatically load more events" button would not be shown.

**Comment by mariusandra** [comment]
Meh. I didn't like how the page flickered when changing queries (screencast above). Updated now (screencast below) and ready for a re-review.

File: frontend/src/queries/schema.ts [hunk 2]
```diff
@@ -204,6 +204,8 @@ export interface DataTableNode extends Node {
     showElapsedTime?: boolean
     /** Show a button to configure the table's columns if possible */
     showColumnConfigurator?: boolean
+    /** Shows a list of saved queries */
```
Comment (pauldambra) [inline frontend/src/queries/schema.ts:207]: nit: you could live without the comment here 😅

**Review by pauldambra — APPROVED** [review]
🚢

**Comment by mariusandra** [comment]
I'm going to use @pauldambra 's ✅ as an excuse to move fast and iterate

**Pull request merged**